### ISSUE TITLE
Warnings from ruby 2.4.0

### ISSUE THREAD

**Issue by ioquatix** [jcupitt/ruby-vips#101]
```
ruby-vips-1.0.4/lib/vips.rb:120: warning: assigned but unused variable - argv
ruby-vips-1.0.4/lib/vips/call.rb:19: warning: ambiguous first argument; put parentheses or a space even after `-' operator
ruby-vips-1.0.4/lib/vips/image.rb:325: warning: shadowing outer local variable - x
```

**Comment by ioquatix** [comment]
The commit I made won't fix the issue, but I thought I'd suggest it as an option. What's the minimum version of Ruby you are targeting with ruby-vips?

**Comment by jcupitt** [comment]
2.0 I think, so it can work on macOS with Apple's Ruby. I should add 2.0 to the travis matrix.

**Comment by ioquatix** [comment]
So, we can use keyword arguments. Not only are they more precise, they are more efficient too, not just in the passing, but avoiding stuff like checking if the last argument is a hash, etc.

**Comment by jcupitt** [comment]
Sure, sounds good. That sounds like a larger change though, so let's have a separate issue for that.

**Comment by ioquatix** [comment]
Well, the change would naturally fix these issues, but I guess we could do that separately - you have the warnings, you could do it pretty easily, then it would be in line with your existing desires for how to change the code. What do you think?

**Comment by jcupitt** [comment]
I've fixed the warnings (I think) and added 2.4 to the test matrix. #102 is the keyword arg issue.

Thanks for pointing these warnings out @ioquatix

**Comment by ioquatix** [comment]
@jcupitt Thanks for being so prompt, you are awesome :)
